scripts/auto-rebase/rebase.sh

```diff
@@ -43,6 +42,159 @@ title() {
     echo -e "\E[34m$1\E[00m";
 }
 
+# LVMS is not integrated into the ocp release image, so the work flow does not fit with core component rebase.  LVMS'
+# operator bundle is the authoritative source for manifest and image digests.
+download_lvms_operator_bundle(){
+    title 'downloading LVMS operator bundles'
+    declare -A images=(["amd64"]="${1}" ["arm64"]="${2:-}")
+    local LVMS_STAGING="${STAGING_DIR}/lvms"
+    for arch in "${!images[@]}"; do
+        # TODO special case arm64 builds of topolvm since they are not available in 4.12
+        if [[ "$arch" == "arm64" && "${images[${arch}]}" == "" ]]; then
+            continue
+        fi
+        img="${images[$arch]}"
+        mkdir -p "$LVMS_STAGING/$arch"
+        pushd "$LVMS_STAGING/$arch" || return 1
+        oc image extract \
+            --registry-config "$PULL_SECRET_FILE" \
+            --path /manifests/:. "$img" \
+            --filter-by-os "$arch" \
+            ||  {
+                    popd
+                    return 1
+                }
+        popd || return 1
+      done
+}
+
+parse_images() {
+    local src=$1
+    yq '.spec.relatedImages[]? | [.name, .image] | @csv' $src > "$(dirname "$src")/images"
+}
+
+write_lvms_images_for_arch(){
+
+    declare -a include_images=(
+        "topolvm-csi"
+        "topolvm-csi-provisioner"
+        "topolvm-csi-resizer"
+        "topolvm-csi-registrar"
+        "topolvm-csi-livenessprobe"
+    )
+
+    local arch="$1"
+    [ -d "$arch" ] || {
+        echo "dir $arch not found"
+        return 1
+    }
+    local csv_manifest="${arch}/lvms-operator.clusterserviceversion.yaml"
+
+    parse_images "$csv_manifest"
+    image_file="./${arch}/images"
+    if ! [ -e $image_file ]; then
+        echo "image file \"$image_file\" not found"
+        return 1
+    fi
+    while read -ers LINE; do
+        name=${LINE%,*}
+        img=${LINE#*,}
+        found=false
+        for included in "${include_images[@]}"; do
+            if [[ "$name" == $included ]]; then
+                found=true
+                break;
+            fi
+        done
+        $found || continue # name not ’found in $REPOROOT/assets/release/release-$arch.json, skip
```

Comment:
Should we log a warning message here? Is this a "normal" situation?

Reply:
This check exists to filter out topolvm images used in OCP (like snapshotter) that are not used in microshift.  It's using the current release-*.json images list as the source of truth for this.